packages/components/psammead-image/src/index.jsx

```diff
@@ -32,13 +32,45 @@ const StyledImg = styled.img`
   width: 100%;
 `;
 
+const getMimeType = srcset => {
+  if (!srcset) return null;
+
+  const firstSrcsetUrl = srcset.split(',')[0].split(' ')[0];
+  const urlFileExtension = firstSrcsetUrl.split('.').pop();
+
+  switch (urlFileExtension) {
+    case 'webp':
+      return 'image/webp';
+    case 'jpg':
+    case 'jpeg':
+      return 'image/jpeg';
+    case 'png':
+      return 'image/png';
+    case 'gif':
+      return 'image/gif';
+    default:
+      return null;
+  }
+};
+
 export const Img = props => {
   const { src, srcset, fallbackSrcset, onLoad, ...otherProps } = props;
 
+  const primaryMimeType = getMimeType(srcset);
+  const secondaryMimeType = getMimeType(fallbackSrcset);
+
   return (
     <StyledPicture onLoad={onLoad}>
-      <source srcSet={srcset} />
-      {fallbackSrcset && <source srcSet={fallbackSrcset} />}
+      <source
+        srcSet={srcset}
+        {...(primaryMimeType && { type: primaryMimeType })}
```

Comment:
not sure you need to conditionally spread like this. I think if the value is null then the attribute doesn't get rendered